skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -74,101 +74,154 @@ object RebalanceCallback extends RebalanceCallbackInstances with RebalanceCallba
 
     def flatMap[B](f: A => RebalanceCallback[F, B]): RebalanceCallback[F, B] = Bind(() => self, f)
 
+    def handleErrorWith(f: Throwable => RebalanceCallback[F, A]): RebalanceCallback[F, A] =
+      HandleErrorWith(() => self, f)
+
     def run(
       consumer: RebalanceConsumer
     )(implicit fToTry: ToTry[F]): Try[A] = {
-      type S = Any => RebalanceCallback[F, Any]
+      type S = Try[Any] => Try[RebalanceCallback[F, Any]]
 
       @tailrec
-      def loop[A1](c: RebalanceCallback[F, A1], ss: List[S]): Try[Any] = {
+      def loop[A1](c: Try[RebalanceCallback[F, A1]], ss: List[S]): Try[Any] = {
         c match {
-          case c: Pure[A1] =>
+          case f: Failure[_] =>
             ss match {
-              case Nil     => c.a.pure[Try]
-              case s :: ss => loop(s(c.a), ss)
+              case Nil     => f
+              case s :: ss => loop(s(f), ss)
             }
-          case c: Lift[F, A1] =>
-            c.fa.toTry match {
-              case Success(a) =>
+          case Success(c) =>
```

Comment:
could you please move unmake of `Success` up, as we expect this to be a happy path

Reply:
done!